### PR TITLE
Hide Sharing settings when Jetpack Sharing module is disabled

### LINKED ISSUE #7160: Sharing menu and options appear even though the module is deactivated.
### Expected behavior
The app should match the module settings on the web. `p4a5px-282-p2`
### Actual behavior
Steps to reproduce:

### Steps to reproduce the behavior
1. Go to the old Jetpack > Modules page in your dashboard.
https://yourjetpacksite.com/wp-admin/admin.php?page=jetpack_modules
2. Deactivate the Sharing module.
3. Open the WordPress app.
4. The Sharing menu and its options are still there, and don’t give any indication aout the buttons not appearing anywhere.

##### Tested on Sony Xperia z3 compact, Android WPAndroid alpha 93

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #7160 
- Show notification that Sharing Jetpack module is disabled and disable sharing options.
- some additional grooming of Jetpack modules
 
To test:
1. Go to the old Jetpack > Modules page in your dashboard.
https://yourjetpacksite.com/wp-admin/admin.php?page=jetpack_modules
2. Deactivate the Sharing module.
3. Open the WordPress app.
4. Go to Sharing > Manage
5. Wait for the Jetpack settings to sync
6. Notification is visible and Sharing settings is hidden.

screenshot_1519738127

### CODE REVIEW

**Comment by oguzkocer** [comment]
@planarvoid Do you mind fixing the merge conflict before I start reviewing this? Thanks!

**Comment by planarvoid** [comment]
@oguzkocer done!

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeButtonPrefsFragment.java [hunk 1]
```diff
@@ -322,6 +328,11 @@ public void run() {
     private void setPreferencesFromSiteSettings() {
         assignPrefListeners(false);
         try {
+            if (!mSiteSettings.isSharingModuleEnabled()) {
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeButtonPrefsFragment.java]: Do you think it'd be better to set the visibility of `mSharingDisabledNotification` and `mSharingSettingsWrapper` a bit more dynamically? So for example, if we decided to add pull to refresh to this page and the user enables sharing after the page is loaded (so the notification is visible) this wouldn't actually hide the notification and show the settings.

It wouldn't actually do anything right now, but will future proof the code a bit. What do you think?

Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeButtonPrefsFragment.java]: I can totally do that. I think I had that implemented at one point in time 👍

**Pull request merged**